### PR TITLE
Support passing in SSLContext

### PR DESCRIPTION
**Pull request by DMaxfield-BDS**
When getting certificates while retrieving the Signature Scanner was not using proxy settings when retrieving the BD certificate.

### CODE REVIEW

File: src/main/java/com/synopsys/integration/rest/client/IntHttpClient.java [hunk 1]
```diff
@@ -279,14 +293,12 @@ private void addBuilderCredentialsProvider() {
...
     private void addBuilderSSLContext() {
         try {
-            SSLContext sslContext;
             HostnameVerifier hostnameVerifier;
             if (alwaysTrustServerCertificate) {
                 logger.error("Automatically trusting server certificates - not recommended for production use.");
                 sslContext = SSLContextBuilder.create().loadTrustMaterial(new TrustAllStrategy()).build();
                 hostnameVerifier = new NoopHostnameVerifier();
             } else {
-                sslContext = SSLContexts.createDefault();
```
Comment (jamesrichard91) [inline src/main/java/com/synopsys/integration/rest/client/IntHttpClient.java:289]: What is the goal of passing in the sslContext? When the `alwaysTrustServerCertificate` it seems like that would ruin the purpose of the change you have made. It seems like when `alwaysTrustServerCertificate` is true you would need to take the sslContext provided and use it to create a new sslContext that will have the `TrustAllStrategy()`

Comment (ekerwin) [reply src/main/java/com/synopsys/integration/rest/client/IntHttpClient.java:289]: We're all over this, thanks James!

File: src/main/java/com/synopsys/integration/rest/client/IntHttpClient.java [hunk 2]
```diff
@@ -93,12 +95,32 @@
...
     public IntHttpClient(IntLogger logger, int timeoutInSeconds, boolean alwaysTrustServerCertificate, ProxyInfo proxyInfo) {
-        this(logger, timeoutInSeconds, alwaysTrustServerCertificate, proxyInfo, new BasicCredentialsProvider(), HttpClientBuilder.create(), RequestConfig.custom(), new HashMap<>());
+        this(logger, timeoutInSeconds, alwaysTrustServerCertificate, proxyInfo, SSL_CONTEXT_SUPPLIER.get());
+    }
+
+    public IntHttpClient(IntLogger logger, int timeoutInSeconds, ProxyInfo proxyInfo, SSLContext sslContext) {
+        this(logger, timeoutInSeconds, false, proxyInfo, sslContext);
+    }
+
+    private IntHttpClient(IntLogger logger, int timeoutInSeconds, boolean alwaysTrustServerCertificate, ProxyInfo proxyInfo, SSLContext sslContext) {
```
Comment (ekerwin) [inline src/main/java/com/synopsys/integration/rest/client/IntHttpClient.java:108]: let's move all private constructors next to each other so all public constructors can be together

File: src/main/java/com/synopsys/integration/rest/client/IntHttpClient.java [hunk 3]
```diff
@@ -414,4 +439,8 @@ public IntLogger getLogger() {
         return logger;
     }
 
+    public SSLContext getSSLContext() {
```
Comment (ekerwin) [inline src/main/java/com/synopsys/integration/rest/client/IntHttpClient.java:442]: I'm not in love with this - do we have to expose this?

**Pull request merged**